<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Soren Harward &lt;stharward@gmail.com&gt;</author>
 <class>PlaylistLengthEditWidget</class>
 <widget class="QWidget" name="PlaylistLengthEditWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>236</width>
    <height>113</height>
   </rect>
  </property>
  <property name="whatsThis">
   <string>An editor for a Playlist Length constraint.  The user can specify the target number of tracks and whether the playlist contain fewer than, equal to, or more than that number of tracks; and the strictness with which the APG should match the target number of tracks.</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Playlist Length Constraint Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QComboBox" name="comboBox_Comparison">
          <property name="toolTip">
           <string/>
          </property>
          <property name="whatsThis">
           <string>Whether the number of tracks in the generated playlist should be less than, equal to, or more than the specified value.</string>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>fewer than</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>equal to</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>more than</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="0" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QSpinBox" name="spinBox_Length">
            <property name="whatsThis">
             <string>The target number of tracks for the generated playlist.</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>30</number>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_Strictness">
          <property name="whatsThis">
           <string>How strict the APG should be about matching the playlist duration to the specified time.</string>
          </property>
          <property name="text">
           <string>Match:</string>
          </property>
          <property name="buddy">
           <cstring>slider_Strictness</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_Fuzzy">
            <property name="text">
             <string>fuzzy</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="slider_Strictness">
            <property name="toolTip">
             <string/>
            </property>
            <property name="whatsThis">
             <string>How strict the APG should be about matching the playlist length to the specified number of tracks.</string>
            </property>
            <property name="maximum">
             <number>10</number>
            </property>
            <property name="value">
             <number>8</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_Exact">
            <property name="text">
             <string>exact</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="2" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>comboBox_Comparison</tabstop>
  <tabstop>slider_Strictness</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
